platform/ios/app-swift/Sources/LocationPrivacyExample.swift

```diff
@@ -9,75 +9,114 @@ enum LocationAccuracyState {
     case fullAccuracy
 }
 
-class Coordinator: NSObject, MLNMapViewDelegate {
-    @Binding var mapView: MLNMapView
-    @Binding var locationAccuracy: LocationAccuracyState
-    var pannedToUserLocation = false
-
-    init(mapView: Binding<MLNMapView>, locationAccuracy: Binding<LocationAccuracyState>) {
-        _mapView = mapView
-        _locationAccuracy = locationAccuracy
+class MapViewModel: NSObject, ObservableObject {
+    // Weak because the coordinator has a strong reference to the view model
+    weak var mapCoordinater: MapLibreRepresentableCoordinator?
+    @MainActor @Published var locationAccuracy: LocationAccuracyState = .unknown
+
+    @MainActor func requestTemporaryLocationAuthorization() {
```

Comment:
Instead of an function here, which is event driven, why not have a @State for the mapview that users can change, called showTemporaryLocationAuthorization? When this state changes, updateUIView would be called, including a reference to MLNMapview.

Reply:
💯 Agreed, updated.